### PR TITLE
Update polaris-form-layout component

### LINKED ISSUE #200: Update polaris-form-layout component
Diff [here](https://github.com/Shopify/polaris/compare/v2.2.0...v2.11.0#diff-2c66129801b5f27907be29599dfc4cd9R2)

### PR DESCRIPTION
**Pull request by vladucu**
fix #200 

Diff [here](https://github.com/Shopify/polaris/compare/v2.2.0...v2.11.0#diff-2c66129801b5f27907be29599dfc4cd9R2)

### CODE REVIEW

File: addon/templates/components/polaris-form-layout/group.hbs [hunk 1]
```diff
@@ -9,3 +13,7 @@
     {{text}}
   {{/if}}
 </div>
+
+{{#if helpText}}
+  <div id={{helpTextID}} class="Polaris-FormLayout__HelpText" data-test-form-layout-group-helpText>{{render-content helpText}}</div>
```
Comment (tomnez) [inline addon/templates/components/polaris-form-layout/group.hbs]: nit: Do we want to break up the element blocks here for easier reading?

```hbs
<div id={{helpTextID}} class="Polaris-FormLayout__HelpText" data-test-form-layout-group-helpText>
  {{render-content helpText}}
</div>
```

If so, same with the title. But just a suggestion.

File: addon/utils/id.js [hunk 2]
```diff
@@ -22,3 +22,8 @@ export function helpTextId(id) {
   return `${id}HelpText`;
 }
 export const computedHelpTextId = makeComputedIdGenerator(helpTextId);
+
+export function idVariation(id, suffix) {
+  return `${id}${suffix}`;
+}
+export const computedIdVariation = makeComputedIdGenerator(idVariation);
```
Comment (andrewpye) [inline addon/utils/id.js]: Now we've got this, how about refactoring the other helpers above to use `idVariation` instead so all of these utils go through the same function?

File: addon/components/polaris-form-layout/group.js [hunk 3]
```diff
@@ -28,18 +35,58 @@ export default Component.extend({
...
    * @private
    */
   role: 'group',
 
+  'data-test-form-layout-group': true,
+
+  titleID: computed('title', function() {
+    if (isBlank(this.get('title'))) {
+      return;
+    }
```
Comment (andrewpye) [inline addon/components/polaris-form-layout/group.js]: Do you think it's worth wrapping up this logic in the `idVariation` util to clean these CPs up? I can't see that we'd ever want to generate an ID variation if the ID isn't present.

Comment (andrewpye) [reply addon/components/polaris-form-layout/group.js]: ooooh yeah, my bad! Plz ignore + continue as you were 😅

**Pull request merged**